src/eddymotion/model/base.py

```diff
@@ -166,12 +177,15 @@ def fit(self, data, n_jobs=None, **kwargs):
 
         self._model = None  # Preempt further actions on the model
 
-    def predict(self, gradient, **kwargs):
+    def predict(self, **kwargs):
```

Comment:
Predict requires you tell the model which index you want to infer.

When models were DWI-only, this had the `gradient`, so you pass the direction. But, if it needs to generalize, I like your idea of having the gradient table store somewhere and just pass the index you want to simulate:

```suggestion
    def predict(self, index, **kwargs):
```

Reply:
OK. See comments in commit df51602. Resolving.